pkg/scheduler/api/shared_device_pool.go

```diff
@@ -0,0 +1,25 @@
+package api
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	GPUSharingDevice = "GpuShare"
+)
+
+type SharedDevicePool interface {
```

Comment:
The interface covers the gpu number as well and npu in the future. It's better to change the name to from `SharedDevicePool` to `Devices`.